openadapt/tools/layout.py

```diff
@@ -0,0 +1,32 @@
+from transformers import pipeline
+from PIL import Image
+from typing import Optional
+import numpy as np
+
+
+class LayoutExtractionTool:
+    def document_query(
+        image: Optional[np.ndarray], image_path: Optional[str], question: str
+    ) -> str:
+        assert (
+            image or image_path
+        ), "At least one of image or image_path must be supplied"
+        assert not (
+            image and image_path
+        ), "Only one of image or image_path may be supplied"
+
+        if image_path:
+            document_image = Image.open(image_path).convert("RGB")
+        else:
+            document_image = image
+
+        query_pipeline = pipeline(
+            "document-question-answering", model="impira/layoutlm-document-qa"
+        )
+
+        output = query_pipeline(document_image, question)
+        if not output:
+            raise TypeError(
+                "Unsupported document type, please input a text-based document"
```

Comment:
This assumes that the model is infallible, which we know is not the case.

How about simply returning `output` and letting the caller decide how to handle it?

Reply:
Sounds good to me, although I would like to raise a warning of some kind to use this exclusively on text based documents.

Reply:
@FFFiend how about:

```
output = query_pipeline(document_image, question)
if not output:
    logger.warning("{output=}")
return output
```